docs/advanced-features/tabulator.md

```diff
@@ -0,0 +1,134 @@
+You can even use named capture groups to extract additional columns from
+the logical key and package name.
+
+The configuration is written in YAML and managed using the
+`quilt3.admin.tabulator`
+[APIs](../api-reference/Admin.md#quilt3.admin.tabulator) or via the
+Quilt Admin UI:
+
+![Admin UI for setting Tabulator configuration](../imgs/admin-tabulator-config.png)
+
+## Configuration
+
+Each Tabulator configuration is written in YAML, following the structure
+outlined below.
+
+### Example
+
+```yaml
+schema:
+  - name: name  # must match ^[a-z_][a-z0-9_]*$
+    type: STRING  # usually BOOLEAN, INT, FLOAT, DOUBLE, STRING, DATE, TIMESTAMP
+  - name: length
+    type: INT
+  - name: effective_length
+    type: FLOAT
+  - name: tpm
+    type: FLOAT
+  - name: num_reads
+    type: FLOAT
+source:
+  type: quilt-packages  # currently the only supported type
+  package_name: "^ccle/(?<date>[^_]+)_(?<study_id>[^_]+)_nfcore_rnaseq$"
+  logical_key: "salmon/(?<sample_id>[^/]+)/quant*\\.genes\\.sf$"
+parser:
+  format: csv  # or `parquet`
+  delimiter: "\t"
+  header: true
+```
+
+A more complex example is available in [this yaml file](./tabulator-example.yaml).
+
+1. **Schema**: The schema defines the columns in the table. Each column must
+   have a name and a type. The name must match the regular expression
+   `^[a-z_][a-z0-9_]*$`.  It does not need to match the column names in the
+   document.  However, if the column names are present in the document, you must
+   set `header` to `true` in the parser configuration.
+2. **Types**:  Must be uppercase and match the [Apache Arrow Data
+Types](https://github.com/awslabs/aws-athena-query-federation/tree/master/athena-federation-sdk#datatypes)
+used by Amazon Athena.  Valid types are BOOLEAN, TINYINT, SMALLINT, INT, BIGINT,
+FLOAT, DOUBLE, STRING, BINARY, DATE, TIMESTAMP.
+3. **Source**: The source defines the packages and objects to query. The `type`
+   must be `quilt-packages`. The `package_name` is a regular expression that
+   matches the package names to include. The `logical_key` is a regular
+   expression that matches the keys of the objects to include. The regular
+   expression may include named capture groups that will be added as columns
+   to the table.
+4. **Parser**: The parser defines how to read the files. The `format` must be
+   one of `csv` or `parquet`. The optional `delimiter` is the character used to
+   separate fields in the CSV file. The optional `header` field is a boolean
+   that indicates whether the first row of the CSV file contains column names.
+
+### Added columns
+
+In addition to the columns defined in the schema, Tabulator will add:
+
+- any named capture groups from the logical key regular expression
+- `$pkg_name` for the package name
+- `$logical_key` for the object as referenced by the package
+- `$physical_key` for the underlying S3 URI
+- `$top_hash` for the revision of the package containing the object (which is
+  currently always `latest`)
```

Comment:
this *might* make an impression that `top_hash = 'latest'`
I *think* it could be written in more straightforward way e.g. "(currently we query only latest package revisions)"